scrapy/exporters.py

```diff
@@ -324,14 +324,14 @@ def _serialize_value(self, value):
             return dict(self._serialize_item(value))
         elif is_listlike(value):
             return [self._serialize_value(v) for v in value]
-        encode_func = to_bytes if self.binary else to_unicode
+        encode_func = bytes if self.binary else str
```

Comment:
I don't think `bytes` works the same way as to_bytes for binary data. I.e. if value is bytes, previously it would have been passed through, but now it raises an exception. The same applies for str.